Tasks/google-play-release/GooglePlay.ts

```diff
@@ -42,54 +42,86 @@ interface GlobalParams {
     params?: PackageParams;
 }
 
-function run() {
+interface Edit {
+    id: string;
+    expiryTimeSeconds: string;
+}
+
+interface Apk {
+  versionCode: number;
+  binary: {
+    sha1: string;
+  };
+}
+
+interface Track {
+  track: string;
+  versionCodes: number[];
+  userFraction: number;
+}
+
+async function run() {
     try {
         tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+        tl.debug('Prepare task inputs.');
+
         let authType: string = tl.getInput('authType', true);
         let key: ClientKey = {};
         if (authType === 'JsonFile') {
-            let serviceAccountKeyFile: string = tl.getPathInput('serviceAccountKey', false);
-            if (!serviceAccountKeyFile) {
-                throw new Error(tl.loc('JsonKeyFileNotFound'));
-            }
+            let serviceAccountKeyFile: string = tl.getPathInput('serviceAccountKey', true, true);
+
             let stats: fs.Stats = fs.statSync(serviceAccountKeyFile);
             if (stats && stats.isFile()) {
                 key = require(serviceAccountKeyFile);
             } else {
-                console.error(tl.loc('InvalidAuthFile'));
-                throw new Error(tl.loc('InvalidAuthFilewithName', serviceAccountKeyFile));
+                tl.debug(`The service account file path ${serviceAccountKeyFile} points to a directory.`);
+                throw new Error(tl.loc('InvalidAuthFile', serviceAccountKeyFile));
             }
         } else if (authType === 'ServiceEndpoint') {
-            let serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), true);
-            if (!serviceEndpoint) {
-                throw new Error(tl.loc('EndpointNotFound'));
-            }
+            let serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), false);
             key.client_email = serviceEndpoint.parameters['username'];
             key.private_key = serviceEndpoint.parameters['password'].replace(/\\n/g, '\n');
         }
 
-        let apkFile: string = resolveGlobPath(tl.getPathInput('apkFile', true));
-        let apkFileList: string[] = [apkFile];
-        let additionalApks: string[] = tl.getDelimitedInput('additionalApks', '\n');
-        if (additionalApks.length > 0) {
-            for (let i = 0; i < additionalApks.length; i++) {
-                apkFileList.push(resolveGlobPath(additionalApks[i]));
-            }
+        let mainApkPattern = tl.getPathInput('apkFile', true);
```

Comment:
getPathInput(...) is defined as function getPathInput(...): string. So, an explicit variable type spec is not needed. ;-)